app.js

```diff
@@ -0,0 +1,130 @@
+/*
+    *** Conventions for this program ***
+
+    *** For the POST API ***
+        DATA FORMAT in JSON is {
+                                    "data": [1,2,3,4,5......]
+                               }
+
+    *** For the PATCH API ***
+        DATA FORMAT in JSON is {
+                                    "num": 5
+                               }
+
+    *** Tested the API calls in POSTMAN
+
+
+*/
+
+const express = require('express');
+const app = express();
+app.use(express.json());
+
+const port = 3000;
+const dataSize = 500;
+
+let numArray = [];
+
+/* 
+    Function to find the index where the new number is to be inserted
+    
+    Parameters: number - number to be inserted
+                low    - the low index for the current search
+                high   - the high index for the current search
+    
+    return value: the index to be inserted at
+*/
+const binarySearch = (number, low, high) => {
+    if (high <= low)
+        return (number > numArray[low])? (low + 1): low;
+    
+    let mid = Math.ceil((low + high) / 2);
+
+    if (number == numArray[mid])
+        return mid + 1;
+
+    if (number > numArray[mid])
+        return binarySearch(number, mid + 1, high)
+    
+    return binarySearch(number, low, mid -1);
+
+}
+
+// GET API 
+const sortNumbers = (req, res) => {
+    if (numArray.length == 0) {
+        return res.status(404).json({
+            status: 'fail',
+            message: 'Emtpy Array'
+        })
+    }
+
+    numArray.sort(function(a,b) {return a -b});
+
+    res.status(202).json({
+        status: 'success',
+        data: numArray
+    });
+
+}
+
+// POST API
+const getNumbers = (req, res) => {
+    let data = req.body;
+    data = data.data;
+
+    if (data.length != dataSize) {
+        return res.status(404).json({
+            status: 'fail',
+            message: 'Input length is not equal to 500'
+        });
+    }
+
+    for (index in data) {
+        if (isNaN(data[index])) {
+            numArray = [];

```

Comment:
If I submit a valid list of numbers, and shortly after someone submits an invalid list of numbers with a NaN, does that mean my original list of valid numbers is gone? (line 85)

Reply:
Given the current implementation, an invalid input would empty the valid list of original numbers. To solve this, we need to remove line number 85. Then, we need to create a local array (say temp = [] ) in the getNumbers function which would store temporary results from the current POST request and if all the numbers in the list checks out to be valid numbers, we would append the temp array to the numArray just below the for loop.